skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from numpy.lib.stride_tricks import as_strided
+
+from skimage.util import invert, view_as_windows
+from ._rolling_ball_cy import apply_kernel, apply_kernel_nan, apply_kernel_flat
+
+
+def rolling_ellipsoid(image, kernel_size=(100, 100), intensity_vertex=100,
+                      has_nan=False):
+    """Estimate background intensity using a rolling ellipsoid.
+
+    The rolling ellipsoid algorithm estimates background intensity for a
+    grayscale image in case of uneven exposure. It is a generalization of the
+    frequently used rolling ball algorithm [1]_.
+
+    Parameters
+    ----------
+    image : ndarray
+        The last two dimensions are treated as the columns and rows of the
+        image. The operation is broadcasted along the remaining ones.
+    kernel_size: two-element tuple, numeric, optional
+        The length of the spatial vertices of the ellipsoid.
+    intensity_vertex : scalar, numeric, optional
+        The length of the intensity vertex of the ellipsoid.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removed.
+
+    Notes
+    -----
+
+    - For the pixel that has its background intensity estimated (at ``(0,0)``)
+      the rolling ellipsoid method places an ellipsoid under it and
+      raises the ellipsoid until it touches the image umbra at ``pos=(y,x)``.
+      The background intensity is then estimated using the image intensity at
+      that position (``image[y, x]``) plus the difference of
+      ``intensity_vertex`` and the intensity of the ellipsoid at ``pos``. The
+      intensity of the ellipsoid is computed using the canonical ellipsis
+      equation::
+
+            semi_spatial = kernel_size / 2
+            semi_vertex = intensity_vertex / 2
+            np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
+
+    - This algorithm assums that low intensity values (black) corresponds to
+      the background. If you have a light background, invert the image before
```

Comment:
```suggestion
      the background. If you have a white background, invert the image before
```
or "high-intensity values" since we just read "low-intensity values (black)" above (alternatively, dark/bright as in the narrative docs)